Remove OTSDIR from path to .password-store

DanHugoDanHugo opened the pull request:
Use only the existing password_store_root() to construct a path to the .password-store used by /usr/bin/pass around which this code wraps.

The user can create a .password-store in a workspace directory and check out the OTS store there without conflicting with an existing ~/.password-store, which is still used IF the PASSWORD_STORE_DIR environment variable used by pass is not set. 

This avoids the specific issue wherein a path to .password-store/USERS was constructed using the OTSDIR environment variable rather than honoring PASSWORD_STORE_DIR.

kfogel commented:
The change looks good to me.  Thanks, @DanHugoDanHugo.

I just committed a `CONTRIBUTING.md` file with commit message guidelines.  Could you revamp the commit message to match those guidelines (I realize this is totally unfair, since the doc wasn't there yet when you made the PR!) and then we'll merge?

kfogel commented:
(It's fine to force-push to the branch as far as I'm concerned, btw.)

DanHugoDanHugo commented:
Yes. As it is, I mistakenly left $, and thus the variable expansion, in my commit message once, a little iteration isn't a bad thing.

kfogel commented:
Ping, @DanHugoDanHugo :-).  Still waiting for the branch update.

Commit:
Remove OTSDIR from path to .password-store

Use PASSWORD_STORE_DIR environment as path to .password-store for pass
if set, or default location if not. This prevents collision with any
existing use of pass in user home and uses the supported pass
environment variable as intended.

kfogel commented:
Thanks, @DanHugoDanHugo!  Looks good to me -- merged.

(I did the merge from the command-line because GitHub is apparently having hiccups right now: it's displaying the "Checking for ability to merge automatically… Hang in there while we check the branch’s status." message forever and never making the green "Merge" button available.  I'm not gonna let some janky web UI block our ability to get a change merged :-). )
